crates/sui-package-alt/src/sui_flavor.rs

```diff
@@ -21,16 +26,103 @@ impl MoveFlavor for SuiFlavor {
 
     type PublishedMetadata = (); // TODO
 
-    type EnvironmentID = String; // TODO
-
     type AddressInfo = (); // TODO
 
     type PackageMetadata = (); // TODO
 
     fn implicit_deps(
-        &self,
-        environments: impl Iterator<Item = Self::EnvironmentID>,
-    ) -> DependencySet<PinnedDependencyInfo> {
-        todo!()
+        file_handle: FileHandle,
+        environments: impl Iterator<Item = EnvironmentID>,
+    ) -> DependencySet<CombinedDependency> {
+        let mut deps = DependencySet::new();
+        let deps_to_skip = ["DeepBook".to_string()];
+        let packages = latest_system_packages();
+        let sha = &packages.git_revision;
+        for env in environments {
+            let pkgs = packages
+                .packages
+                .iter()
+                .filter(|package| !deps_to_skip.contains(&package.package_name));
+
+            for package in pkgs {
+                let repo = SYSTEM_GIT_REPO.to_string();
+                let dep = ManifestDependencyInfo::Git(ManifestGitDependency {
+                    repo: repo.clone(),
+                    rev: Some(sha.clone()),
+                    subdir: PathBuf::from(&package.repo_path),
+                });
+
+                let dep_info = CombinedDependency::new(Dependency::new(
+                    dep,
+                    env.clone(),
+                    None, // TODO: is this correct? We don't have this information
+                    // here
+                    file_handle,
+                    true,
+                ));
+
+                // TODO: should we change the package name to be of tyep Identifier rather than
```

Comment:
we could, although I think maybe they're currently set to capitalized non-identifiers (e.g. `Sui` and `MoveStdlib`). We should maybe change that, but I wonder if it might break things. Maybe a better approach would be to expand the information in the manifest (`manifest.json` in `sui-framework-snapshot` IIRC) to add an identifier (so we would have both `MoveStdlib` and `std` stored, for example).